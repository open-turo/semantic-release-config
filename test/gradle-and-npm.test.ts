import { execSync } from "node:child_process";

jest.mock("node:child_process", () => ({
  execSync: jest.fn(),
}));

describe("gradle-and-npm", () => {
  test("adds gradle-semantic-release-plugin, semantic-release/npm, and semantic-release/git plugins", async () => {
    (execSync as jest.Mock).mockImplementation(() => {});
    const openapi = await import("~/gradle-and-npm");
    const gradlePlugin = openapi.plugins[3];
    const npmPlugin = openapi.plugins[4];
    const openApiPlugin = openapi.plugins[5];
    const semanticReleasePlugin = openapi.plugins[6];
    expect(gradlePlugin).toMatchSnapshot();
    expect(npmPlugin).toMatchSnapshot();
    expect(openApiPlugin).toMatchSnapshot();
    expect(semanticReleasePlugin).toMatchSnapshot();
  });
});
