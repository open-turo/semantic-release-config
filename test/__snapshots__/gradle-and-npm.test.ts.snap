// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gradle-and-npm adds gradle-semantic-release-plugin, semantic-release/npm, and semantic-release/git plugins 1`] = `"gradle-semantic-release-plugin"`;

exports[`gradle-and-npm adds gradle-semantic-release-plugin, semantic-release/npm, and semantic-release/git plugins 2`] = `
[
  "@semantic-release/npm",
  {
    "tarballDir": "pack",
  },
]
`;

exports[`gradle-and-npm adds gradle-semantic-release-plugin, semantic-release/npm, and semantic-release/git plugins 3`] = `
[
  "@semantic-release/git",
  {
    "assets": [
      "README.md",
      "gradle.properties",
      "package-lock.json",
      "package.json",
      "yarn.lock",
    ],
    "message": "ci(release): \${nextRelease.version} <% nextRelease.channel !== 'next' ? print('[skip ci]') : print('') %>

\${nextRelease.notes}",
  },
]
`;

exports[`gradle-and-npm adds gradle-semantic-release-plugin, semantic-release/npm, and semantic-release/git plugins 4`] = `
[
  "@semantic-release/exec",
  {
    "failCmd": "[ -x ./script/semantic-release-fail ] && ./script/semantic-release-fail '\${nextRelease.version}' || true",
    "publishCmd": "[ -x ./script/semantic-release-publish ] && ./script/semantic-release-publish '\${nextRelease.version}' || true",
    "successCmd": "[ -x ./script/semantic-release-success ] && ./script/semantic-release-success '\${nextRelease.version}' || true",
  },
]
`;
